XenAdmin/Dialogs/CallHome/CallHomeSettingsDialog.cs

```diff
@@ -177,23 +190,105 @@ private void enrollmentCheckBox_CheckedChanged(object sender, EventArgs e)
             UpdateButtons();
         }
 
-        private void callHomeAuthenticationPanel1_AuthenticationChanged(object sender, EventArgs e)
+        private void newAuthenticationRadioButton_CheckedChanged(object sender, EventArgs e)
+        {
+            SetMyCitrixCredentials(existingAuthenticationRadioButton.Checked);
+        }
+
+        private void radioButton2_CheckedChanged(object sender, EventArgs e)
+        {
+            SetXSCredentials(currentXsCredentialsRadioButton.Checked);
+        }
+
+        private void SetXSCredentials(bool useCurrent)
         {
-            Program.Invoke(this, delegate
+            if (useCurrent)
+            {
+                textboxXSUserName.Text = pool.Connection.Username;
+                textboxXSPassword.Text = pool.Connection.Password;
+                textboxXSUserName.Enabled = false;
+                textboxXSPassword.Enabled = false;
+            }
+            else
             {
-                if (callHomeAuthenticationPanel1.Authenticated)
-                {
-                    authenticated = true;
-                    authenticationToken = pool.CallHomeSettings.GetExistingSecretyInfo(pool.Connection, CallHomeSettings.UPLOAD_TOKEN_SECRET);
-                }
-                UpdateButtons();
-            });
+                textboxXSUserName.Text = xsUserName;
+                textboxXSPassword.Text = xsPassword;
+                textboxXSUserName.Enabled = true;
+                textboxXSPassword.Enabled = true;
+            }
         }
 
-        private void newAuthenticationRadioButton_CheckedChanged(object sender, EventArgs e)
+        private void SetMyCitrixCredentials(bool useExisting)
+        {
+            if (useExisting)
+            {
+                //textBoxMyCitrixUsername.Text = String.Empty;

```

Comment:
I think it would make sense to show whatever we have (disabled anyway)
For passwords some characters can indicate that something is there.